### DISCUSSION TITLE
Duplicate document content without redirect

### DISCUSSION THREAD

**Discussion by JoelMTaylor** [retypeapp/retype#334] [Ideas]
Hi, 

Thank you for all the work you have been doing. Version 2.3 is great and has many useful additions/fixes.

I was wondering if it is possible to duplicate the content of a page without switching to that page? My testing of _redirect_ resulted in the selected page (in the left navigation bar) changing to the page in the redirect. I am interested in the same functionality (displaying the content from that page), but not changing the navigation. 

I played around with _redirect_ and _route_, but these did not seem to achieve my goal (to be honest, I didn't figure out what _route_ does).

Thanks.

**Comment by geoffreymcgill** [comment]
> I was wondering if it is possible to duplicate the content of a page without switching to that page?

Can you explain this further? maybe describing a sample scenario

It is possible to use includes now. A file such as `_includes/my-snippet.md` would be created and then referenced on any page using the following syntax:

```md
# Sample page

This is a sample page demonstrating includes.

{{ include "my-snippet" }}
```

The final result should look like the following:

<img width="454" alt="Screen Shot 2022-05-26 at 6 21 00 PM" src="https://user-images.githubusercontent.com/62210/170604447-1158913e-57b6-4cad-8e3b-1f6117c83dd4.png">

Hope this helps.

**Reply by JoelMTaylor** [reply]
Yes, that is basically what I am looking for. The only change that I would make (what I would ideally like) is for the page to be in a random folder and not be required to be located in ```_includes```.

However, it does the job of only creating a page once and referencing that page without the navigation bar changing.

Thank you.

**Reply by geoffreymcgill** [reply]
The included file can be located anywhere within the project by just passing the path to the file. 

In the following sample, I moved the `my-snippet.md` file into the same folder as the `sample-page.md`, but also prefixed with an `_` underscore to prevent the included file from being generated as a standalone page.

```md
# Sample page

This is a sample demonstrating includes.

{{ include "_my-snippet" }}

{{ include "_my-snippet.md" }}
```

The `.md` file type suffix is optional if the included files are stored within the `_includes` folder or if the included file is prefixed with an `_` underscore.

If the file was not prefixed with an `_` underscore, Retype will create a separate page (`/my-snippet/`) as well as including the content of that page into `sample-page.md`. The content would be duplicated on `/my-snippet/` and `/sample-page/`, which may also be the authors intention.

Hope this helps.

**Reply by JoelMTaylor** [reply]
This helps significantly. Thank you.

**Comment by geoffreymcgill** [comment]
The [`route`](https://retype.com/configuration/page/#route) is for setting a custom path to a page, instead of the generated path based on the folder/file name. 

For example, you might have a file located at `/guides/introduction.md`, which would create a URL such as `example.com/guides/introduction/`, but you would rather have that page available at `example.com/getting-started/`. 

You can set `route: /getting-started` in the page metadata and now the file stored at `/guides/introduction.md` will be hosted from `example.com/getting-started`.

The most typical scenario I've seen for setting a custom `route` is when you want to support existing URL's from a previous website and want to ensure the existing links in the wild do not break (404). 

Hope this helps.

**Reply by JoelMTaylor** [reply]
Thank you for taking the time to explain routing, too.
